File: diary-server/src/main/java/com/diary/server/config/WeChatProperties.java
package com.diary.server.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "diary.wechat")
public record WeChatProperties(
        String appId,
        String appSecret,
        boolean devMode
) {
    public boolean isConfigured() {
        return appId != null && !appId.isBlank()
                && appSecret != null && !appSecret.isBlank();
    }
}

File: diary-server/src/main/kotlin/com/diary/server/config/WeChatProperties.kt
package com.diary.server.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "diary.wechat")
data class WeChatProperties(
    val appId: String?,
    val appSecret: String?,
    val devMode: Boolean
) {
    val isConfigured: Boolean
        get() = !appId.isNullOrBlank() && !appSecret.isNullOrBlank()
}
